libsolidity/formal/EncodingContext.cpp

```diff
@@ -207,9 +208,43 @@ void EncodingContext::transfer(Expression _from, Expression _to, Expression _val
 		m_balances->valueAtIndex(indexBefore),
 		m_balances->valueAtIndex(indexAfter)
 	);
-	m_solver.addAssertion(m_balances->currentValue() == newBalances);
+	m_solver->addAssertion(m_balances->currentValue() == newBalances);
 }
 
+/// Solver.
+
+Expression EncodingContext::assertions()
+{
+	if (m_assertions.empty())
+		return Expression(true);
+
+	return m_assertions.back();
+}
+
+void EncodingContext::pushSolver()
+{
+	if (m_assertions.empty())
```

Comment:
Why not `m_assertions.push_back(assertions())`?

Reply:
right